lib/Models/Segment.php

```diff
@@ -159,9 +159,11 @@ static function getWooCommerceSegment() {
   static function shouldShowWooCommerceSegment() {
     $woocommerce_helper = new WCHelper();
     $is_woocommerce_active = $woocommerce_helper->isWooCommerceActive();
-    $woocommerce_users_count = Subscriber::where('is_woocommerce_user', 1)->count();
+    $woocommerce_user_exists = Subscriber::where('is_woocommerce_user', 1)
+      ->limit(1)
+      ->findOne();
```

Comment:
I just realized that if you have no WC users this will always result in table full scan due to the fact there's no index on `is_woocommerce_user`(and the fact it will never find one). I guess that's risky for large customers. Should we try sth like below? That should use indexes.

```
Segment::tableAlias('segment')
  ->where('segment.type', Segment::TYPE_WC_USERS)
  ->join(
    MP_SUBSCRIBER_SEGMENT_TABLE,
    'segment_subscribers.segment_id = segment.id',
    'segment_subscribers'
  )
  ->limit(1)
  ->findOne();
```

Reply:
👍 used that, it's working correctly